---
uint32 num_checkpoints
